src/routes/+layout.svelte

```diff
@@ -14,20 +14,42 @@
 
 	const mobileWidth = 700
 
-	const dietePages = {
-		'/': 'Intro',
+	const menu: { [title: string]: { [path: string]: string } } = {
+		Diète: {
+			'/': 'Intro',
+		},
+		Elements: {},
+		'Basic components': {
+			'/components/button': 'Button',
+		},
 	}
```

Comment:
```suggestion
	const menu: { [title: string]: { [path: string]: string } } = {
		Diète: {
			'/': 'Intro',
		},
		Elements: {},
		'Basic components': {
			'/components/button': 'Button',
		},
	} as const
```

Reply:
As far as I understand this won't make any difference in runtime performance, because all it does (in the case of object literals) is to make all the properties `readonly`. However that is entirely a compile time Typescript feature which, when compiled to Javascript, disappears completely. 

Also by providing the typing for `menu` (which is necessary to be able to index the keys as strings) without `readonly` we remove the effect of `const` even in Typescript. So in our case it won't achieve much.

There are some cases when `as const` may help the compiler to deduce the correct type (typically with tuples with fixed number of items) or with not widening string literal types to strings, but unfortunately here neither is the case.

Maybe there is a way that the Svelte compiler can use this information? In that case adding `readonly` to the type of `menu` could be a solution.

Reply:
I agree with all you said. I don't think it helps Svelte compiler in any way. My understanding is, that it is a a readonly object and this will make it clearer, declaring such objects is becoming a convention as it helps detect errors in case someone decides to modify it somewhere in code. But yeah, we don't use this anywhere else, feel free to reject the suggestion.